network.go

```diff
@@ -832,35 +832,84 @@ func (n *network) updateSvcRecord(ep *endpoint, localEps []*endpoint, isAdd bool
 	}
 
 	c := n.getController()
-	sr, ok := c.svcDb[n.ID()]
-	if !ok {
+	c.Lock()
```

Comment:
I thought about that. But since this is a common function that we use today for add and delete, I thought it is okay to leave it here.
If we remove this, then this function really has no value :)